app/src/main/java/com/example/makeitso/MakeItSoMessagingService.kt

```diff
@@ -1,18 +1,68 @@
 package com.example.makeitso
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.NotificationManager.IMPORTANCE_DEFAULT
+import android.app.PendingIntent
+import android.app.PendingIntent.FLAG_IMMUTABLE
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_CLEAR_TOP
+import android.os.Build
 import android.util.Log
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.core.app.NotificationCompat
 import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import kotlin.random.Random
 
 class MakeItSoMessagingService : FirebaseMessagingService() {
-    /**
-     * Called if the FCM registration token is updated. This may occur if the security of
-     * the previous token had been compromised. This is called when the FCM registration
-     * token is initially generated so this is where you would retrieve the token.
-     */
+
+    private val random = Random
```

Comment:
nit: Does it make any difference to declare Random beforehand rather than on the callsite?

Reply:
To avoid recreating the object every time we receive a notification!